components/service/glean/src/main/java/mozilla/components/service/glean/config/DefaultConfiguration.kt

```diff
@@ -0,0 +1,8 @@
+package mozilla.components.service.glean.config
+
+public val DefaultConfiguration = Configuration(
+    serverEndpoint = "https://incoming.telemetry.mozilla.org",
+    userAgent = "Glean/1.0 (Android)",
```

Comment:
"1.0" doesn't match the release version number. Is this something you would like to use? Currently we do not expose the version as a code constant. But we could.

Reply:
@mreid-moz to the rescue! Do you think exposing the Glean version through the "UserAgent" header is of any use? Otherwise I'll gladly morph this to `"Glean (Android"`

Reply:
Generally I think it will be very useful to know the Glean SDK version for a given submission, since it will allow us to handle the possible future case where we want to make changes to the client-server protocol (or server-side routing/processing). Putting it in the `UserAgent` string seems like a good approach.

The current Telemetry ingestion pipeline does not parse or pass along the `UserAgent` string (though support code for that exists), but the next gen [gcp-based ingestion infra](https://github.com/mozilla/gcp-ingestion/blob/b289340aae93fee5f6673cee2e715433f1c50f08/ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParseUserAgent.java) does UA-parsing.

One piece of prior art: when developing PingSender, we added a specific header `X-PingSender-Version`, maybe because we didn't already have UserAgent parsing at the time? If we want to go with this approach, I would prefer to generalize to something like a pair of `X-Client-Type` and `X-Client-Version` headers.

Reply:
> One piece of prior art: when developing PingSender, we added a specific header `X-PingSender-Version`, maybe because we didn't already have UserAgent parsing at the time? If we want to go with this approach, I would prefer to generalize to something like a pair of `X-Client-Type` and `X-Client-Version` headers.

Good point! I didn't remember that, you're right. We added these two headers because the pipeline, at the time, didn't support UserAgent parsing (see [the docs](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/internals/pingsender.html)). I will add the two headers for backwards compatibility with the current pipeline, and still send the UserAgent, as done by the `pingsender` on desktop.

@pocmo turns out we might need to export the version as a code constant. Is this something that's trivial to do? Can I take this as a separate PR (I will file a bug to track this)?

Reply:
Yes, this should be trivial. I can try that and open a PR.

Reply:
Thank you!

Reply:
> > One piece of prior art: when developing PingSender, we added a specific header `X-PingSender-Version`, maybe because we didn't already have UserAgent parsing at the time? If we want to go with this approach, I would prefer to generalize to something like a pair of `X-Client-Type` and `X-Client-Version` headers.
> 
> Good point! I didn't remember that, you're right. We added these two headers because the pipeline, at the time, didn't support UserAgent parsing (see [the docs](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/internals/pingsender.html)). I will add the two headers for backwards compatibility with the current pipeline, and still send the UserAgent, as done by the `pingsender` on desktop.

FWIW we don't currently use the two headers I mentioned, we only use the pingsender-specific one. I just figured that to be future proof, we could use the same thing here and elsewhere. I think sending the version in the `UserAgent` is probably sufficient if we want to go that route.

Reply:
Oh! @mreid-moz , should I remove the 2 additional headers or leave them?